### DISCUSSION TITLE
Allow users (non-admins) the ability to choose what related fields show in table and export

### DISCUSSION THREAD

**Discussion by crhampton1** [directus/directus#5639] [Feature Requests]
Hey! For the past week or so I have been experimenting with seeing if we could use DirectUs. We currently use a non-api content management software called BrickRiver that is growing older and lacks a tons of features Directus has automatically. Directus is an amazing product!!

One of the things we do happen to like about our old system is that in our table results view and exports, we can choose what fields show up even within related fields. An example is:

![Screen Shot 2021-05-12 at 1 41 30 PM](https://user-images.githubusercontent.com/1928925/118020183-cd09cc80-b327-11eb-9f3d-8ad1f0a1f1e3.png)

This pulls the Charge, District, and Churches "collections" (Charge is parent, District and Churches show up as relationships) and we set the name and "id number (gcfa number)" as fields in the related content. The thing is, this is dynamic and changes depending on the user and sometimes even on the query. Every user can dynamically change what they see in the table and then the table automatically is what is downloaded when you export.

From what I can see, this is not currently possible in Directus. I love the more advanced filtering, but we have quite a few users who would use collections (we need more as an internal tool, not a website content manager) and would need to change and export different fields in related content.

It would be amazing if it worked similar to the advanced filtering, where you can go into related collections and select the fields.

Is that possible, on the roadmap, or not going to happen? It's probably the only thing keeping me back from recommending the switch. If not, I will try to think of other solutions to make it work! Thanks!

**Comment by rijkvanzanten** [comment]
See https://github.com/directus/directus/pull/3592 for an in-depth discussion around this functionality 🙂 

As of right now, to achieve what you're looking for, you can use the `related-values` display on the many-to-one field of the current collection. That allows you to chose what columns of the nested m2o field to show in the parent context 👍🏻

**Reply by crhampton1** [reply]
Thanks @rijkvanzanten, sounds like the answer might be no:

"We do not add support for rendering nested relational values on the layouts directly, but instead:"

Is what we needed, vs me as the admin setting every single field up as a virtual field too. Each of our users have hundreds of saved searches and each one they can randomly choose any of the  deeply nested fields (sometimes 3-4 layers deep) on the fly and different filters without any extra setup from admins.

Thanks for your help, though!!

**Reply by rijkvanzanten** [reply]
Makes sense! As with nearly everything within Directus, we're always up to keep discussing 🙂 None of this is set in stone yet 👍🏻

**Reply by rijkvanzanten** [reply]
The nice thing about the virtual field approach is that it works regardless of layout used, where this nested columns setup would only apply to the table layout (cc @benhaynes)

**Reply by benhaynes** [reply]
The previous way of selecting deeply nested fields (the tree-view on relational fields) should work on other Layouts too, right? You could select a relational `category.name` on a Card subtitle, etc... right?

I see the benefit of both selecting deeply nested relational values case-by-case (by user), and the virtual fields (configured by admin). I know we had some issues with the deeply nested selection, but I don't remember exactly what that was. As Rijk said, always down to keep discussing. Also, export could have a different config than Layouts... and could potentially allow selecting the deep fields.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
